reo/tests/test_chp.py

```diff
@@ -215,3 +215,78 @@ def test_chp_resilience_unavailability(self):
 
+    def test_supplementary_firing(self):
+        """
+        validation to ensure that: 
+        (1) CHP Supplementary firing option loads as intended
+        (2) CHP Supplementary firing is used to meet heating load when pricing is low relative to boiler fuel use
+        (3) CHP Supplementary firing is not used to meet heating load when capital cost it too high to justify purchase
+        (4) Capital Cost of CHP Supplementary firing is included in objective function
+        """
+        # Call API, get results in "d" dictionary
+        nested_data = json.load(open(self.test_post, 'rb'))
+        # CHP
+        nested_data["Scenario"]["timeout_seconds"] = 420
+        nested_data["Scenario"]["optimality_tolerance_bau"] = 0.001
+        nested_data["Scenario"]["optimality_tolerance_techs"] = 0.01
+        nested_data["Scenario"]["Site"]["CHP"]["prime_mover"] = "combustion_turbine"
+        nested_data["Scenario"]["Site"]["CHP"]["size_class"] = 2
+        nested_data["Scenario"]["Site"]["CHP"]["min_kw"] = 800
+        nested_data["Scenario"]["Site"]["CHP"]["max_kw"] = 800
+        #Supplementary Firing
+        nested_data["Scenario"]["Site"]["CHP"]["supplementary_firing_capital_cost_per_kw"] = 10000
+        nested_data["Scenario"]["Site"]["CHP"]["supplementary_firing_max_steam_ratio"] = 5.0
+        nested_data["Scenario"]["Site"]["CHP"]["supplementary_firing_efficiency"] = 0.9
+        #Flat Thermal and Electrical Loads
+        kw_to_mmbtu = 0.00340951064
+        nested_data["Scenario"]["Site"]["LoadProfileBoilerFuel"]["loads_mmbtu_per_hour"] = [3000*kw_to_mmbtu]*8760
+        nested_data["Scenario"]["Site"]["LoadProfile"]["loads_kw"] = [800]*8760
+        nested_data["Scenario"]["Site"]["LoadProfile"]["critical_loads_kw"] = [800]*8760
+        #Make Boiler relatively inefficient to justify use of chp supplementary firing when available  
+        nested_data["Scenario"]["Site"]["Boiler"]["boiler_efficiency"] = 0.95
+        # PV set to zero
+        nested_data["Scenario"]["Site"]["PV"]["min_kw"] = 0
+        nested_data["Scenario"]["Site"]["PV"]["max_kw"] = 0
+        # Run, and record capital cost and thermal production
+        resp = self.get_response(data=nested_data)
+        self.assertHttpCreated(resp)
+        r = json.loads(resp.content)
+        run_uuid = r.get('run_uuid')
+        d = ModelManager.make_response(run_uuid=run_uuid)
+        # Edit capital cost to a more reasonable value
+        c = nested_to_flat_chp(d['outputs'])
+        d_expected = {}
+        d_expected['chp_kw'] = 800
+        d_expected['chp_supplementary_firing_kw'] = 0
+        d_expected['chp_year_one_electric_energy_produced_kwh'] = 800*8760
+        d_expected['chp_year_one_thermal_energy_produced_mmbtu'] = 800*(0.4418/0.3573)*(kw_to_mmbtu)*8760
+        try:
+            check_common_outputs(self, c, d_expected)
+        except:
+            print("Run {} expected outputs may have changed. Check the Outputs folder.".format(run_uuid))
```

Comment:
We have not had an outputs folder for a long time. I know that this is just a copy and paste but we should get rid of these messages throughout the tests. In fact, I noticed that you added to the `nested_to_flat` function for some tests (or maybe Alex did), but I also think that we should deprecate `check_common_outputs`, which will mean that we no longer have to maintain `nested_to_flat`.